test/uk/gov/hmrc/timetopayproxy/models/CreatePlanRequestSpec.scala

```diff
@@ -291,6 +291,20 @@ class CreatePlanRequestSpec extends AnyWordSpec with Matchers with CreatePlanReq
       }
     }
 
+    "fail decoding if addressPostcode is empty" in {
+      import play.api.libs.json._
+
+      Try(
+        Json
+          .parse(getJsonWithInvalidReference(addressPostcode = ""))
+          .validate[CreatePlanRequest]
+      ) match {
+        case Failure(t) =>
+          t.toString() shouldBe "java.lang.IllegalArgumentException: requirement failed: addressPostcode should not be empty"
+        case _ => fail("Response should be a validation error")
```

Comment:
We shouldn't use `_` here as we will want to see what the failure reason is for not passing the test :)

Reply:
Also refactored the test structure so to remove extra matching off of statements like advised in another comment. Let me know what you think.